### DISCUSSION TITLE
Transactional Mail Endpoint with Template Extensions

### DISCUSSION THREAD

**Discussion by benhaynes** [directus/directus#3104] [Feature Requests]
Email Templates (a new type of Extension), and figuring out Role Permissions — who can send emails? Limit recipient? etc.

**Comment by RANSoftRA** [comment]
Hey @benhaynes - I am currently working with the current directus rc (30) - i was wondering if there is already a way to create custom email tempaltes.

The docs mention it, but are currently not providing further information on that topic: https://docs.directus.io/guides/extensions/email-templates.html#full-example

It seems linke the code does not support this at the moment as well:

https://github.com/directus/directus/blob/main/api/src/mail/index.ts

```
const liquidEngine = new Liquid({
	root: path.resolve(__dirname, 'templates'),
	extname: '.liquid',
});
...

export async function sendPasswordResetMail(email: string, url: string) {
	if (!transporter) return;

	const defaultOptions = await getDefaultTemplateOptions();

	const html = await liquidEngine.renderFile('password-reset', {
		...defaultOptions,
		email,
		url,
	});
...
```
Am I missing something or is this feature still under discussion?

Btw: Really love what you guys are doing - i really like working with this piece of software.

**Reply by benhaynes** [reply]
Thanks!!

I know creating a full transactional email service within Directus is still in work, but I believe the email API endpoint is good to go. That, combined with custom API endpoints should allow you to create whatever you need. I don't specifically remember the state of "Email Templates" — I know we have some internally, but I'm not sure about template extensions. @rijkvanzanten ?

Either way, this is something we want to do, and want to do right... so it's good to get a lot of ideas here while we prepare to finish it up.

**Reply by rijkvanzanten** [reply]
> but I believe the email API endpoint is good to go. 

This feature request is the (unresolved) feature request to implement a mail endpoint with custom templates 👀 

---

We don't currently offer custom email templates, or a way to send custom emails. Regarding the endpoint. I think my preferred way to go with this is to offer a "MailService" that you can use in a custom endpoint (that allows you to pass custom html / liquid as well). Having a dedicated /mail endpoint opens up a big rabbit hole in security / permissions, so I'm not sure if we should charge ahead with that without having it thought through fully

**Reply by niklasgrewe** [reply]
any updates on this topic?

**Reply by benhaynes** [reply]
[**Unfortunately, we can not provide estimates on features or fixes**](https://docs.directus.io/getting-started/support.html#can-you-give-an-eta-for-this-feature-fix). As of now, the only way to get an accurate timeline or release date is to [sponsor](https://docs.directus.io/getting-started/support.html#sponsored-work) this task. We recognize that this may be an important feature/fix, but we are a small open-source organization with a lot to triage and complete. If you can't sponsor this ticket, then the next best thing is to increase its priority by giving it a 👍 . Or contribute a pull-request... after all, this is OSS!

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
